src/app/participate/participate.component.ts

```diff
@@ -30,25 +30,27 @@ export class ParticipateComponent implements OnInit {
   depositData: string;
   depositDataFormGroup: FormGroup;
   deposited: boolean|'pending'  = false;
+  depositContractAddress: string;
   readonly BOOTNODE_ADDRESS = environment.bootnodeAddress;
   readonly MIN_BALANCE_IN_ETH = '0.35';
   readonly MIN_BALANCE = toWei(this.MIN_BALANCE_IN_ETH, 'ether');
-  readonly CONTRACT_ADDRESS = DEPOSIT_CONTRACT_ADDRESS;
   readonly DOCKER_TAG = "latest";
 
   @ViewChild('autosize') autosize: CdkTextareaAutosize;
   @ViewChild('stepper') stepper: MatStepper;
 
   constructor(
     private readonly portis: PortisService,
-    private readonly metamask: MetamaskService, 
+    private readonly metamask: MetamaskService,
     private readonly snackbar: MatSnackBar,
     private readonly storage: LocalStorage,
     private readonly progress: ProgressService,
     private readonly faucet: FaucetService,
     private readonly cdr: ChangeDetectorRef,
     private readonly formBuilder: FormBuilder,
-  ) { }
+  ) {
+    this.depositContractAddress = this.web3.depositContractAddress;
```

Comment:
`this.web3` is not defined until ngOnInit, so you are going to want to make this initialization within `chooseWeb3Provider`